app/scripts/views/form.js

```diff
@@ -308,19 +294,20 @@ define(function (require, exports, module) {
         return;
       }
 
-      // exclude coppa inputs from validation, coppa has its own validation
-      var inputEls = this.$('input').not('#coppa input');
+      const inputEls = this.$('input');
       for (var i = 0, length = inputEls.length; i < length; ++i) {
-        var el = inputEls[i];
-        if (! this.isElementValid(el)) {
-          var fieldType = this.getElementType(el);
-
-          if (fieldType === 'email') {
-            return this.showEmailValidationError(el);
-          } else if (fieldType === 'password') {
-            return this.showPasswordValidationError(el);
-          }
+        const el = inputEls[i];
+        const $el = this.$(el);
 
+        let validationError;
+        try {
+          $el.validate();
+        } catch (e) {
+          validationError = e;
+        }
+
+        if (validationError) {
```

Comment:
No big deal, but I'm curious why you made this a separate condition rather than just doing it inside the `catch`. Was it just because you find it cleaner this way or am I missing something more significant?